diff -r d9e357c04dfa Android.mk
--- a/Android.mk	Sun Mar 23 09:51:24 2014 -0700
+++ b/Android.mk	Tue Apr 01 18:45:55 2014 +0200
@@ -11,8 +11,8 @@
 
 # Enable this if you want to support loading MOD music via mikmod
 # The library path should be a relative path to this directory.
-SUPPORT_MOD_MIKMOD := true
-MIKMOD_LIBRARY_PATH := external/libmikmod-3.1.12
+# SUPPORT_MOD_MIKMOD := true
+# MIKMOD_LIBRARY_PATH := external/libmikmod-3.1.12
 
 # Enable this if you want to support loading MP3 music via SMPEG
 # The library path should be a relative path to this directory.
@@ -84,18 +84,49 @@
 ifeq ($(SUPPORT_MP3_SMPEG),true)
     LOCAL_C_INCLUDES += $(LOCAL_PATH)/$(SMPEG_LIBRARY_PATH)
     LOCAL_CFLAGS += -DMP3_MUSIC
-    LOCAL_SHARED_LIBRARIES += smpeg2
+	LOCAL_SRC_FILES += \
+	$(SMPEG_LIBRARY_PATH)/MPEG.cpp \
+	$(SMPEG_LIBRARY_PATH)/MPEGlist.cpp \
+	$(SMPEG_LIBRARY_PATH)/MPEGring.cpp \
+	$(SMPEG_LIBRARY_PATH)/MPEGstream.cpp \
+	$(SMPEG_LIBRARY_PATH)/MPEGsystem.cpp \
+	$(SMPEG_LIBRARY_PATH)/plaympeg.c \
+	$(SMPEG_LIBRARY_PATH)/smpeg.cpp \
+	$(SMPEG_LIBRARY_PATH)/audio/bitwindow.cpp \
+	$(SMPEG_LIBRARY_PATH)/audio/filter.cpp \
+	$(SMPEG_LIBRARY_PATH)/audio/filter_2.cpp \
+	$(SMPEG_LIBRARY_PATH)/audio/hufftable.cpp \
+	$(SMPEG_LIBRARY_PATH)/audio/MPEGaudio.cpp \
+	$(SMPEG_LIBRARY_PATH)/audio/mpeglayer1.cpp \
+	$(SMPEG_LIBRARY_PATH)/audio/mpeglayer2.cpp \
+	$(SMPEG_LIBRARY_PATH)/audio/mpeglayer3.cpp \
+	$(SMPEG_LIBRARY_PATH)/audio/mpegtable.cpp \
+	$(SMPEG_LIBRARY_PATH)/audio/mpegtoraw.cpp \
+	$(SMPEG_LIBRARY_PATH)/video/decoders.cpp \
+	$(SMPEG_LIBRARY_PATH)/video/floatdct.cpp \
+	$(SMPEG_LIBRARY_PATH)/video/gdith.cpp \
+	$(SMPEG_LIBRARY_PATH)/video/jrevdct.cpp \
+	$(SMPEG_LIBRARY_PATH)/video/mmxflags_asm.S \
+	$(SMPEG_LIBRARY_PATH)/video/mmxidct_asm.S \
+	$(SMPEG_LIBRARY_PATH)/video/motionvec.cpp \
+	$(SMPEG_LIBRARY_PATH)/video/MPEGvideo.cpp \
+	$(SMPEG_LIBRARY_PATH)/video/parseblock.cpp \
+	$(SMPEG_LIBRARY_PATH)/video/readfile.cpp \
+	$(SMPEG_LIBRARY_PATH)/video/util.cpp \
+	$(SMPEG_LIBRARY_PATH)/video/video.cpp \
+    # LOCAL_SHARED_LIBRARIES += smpeg2
 endif
 
+
 ifeq ($(SUPPORT_OGG),true)
     LOCAL_C_INCLUDES += $(LOCAL_PATH)/$(OGG_LIBRARY_PATH)/include $(LOCAL_PATH)/$(VORBIS_LIBRARY_PATH)
     LOCAL_CFLAGS += -DOGG_MUSIC -DOGG_USE_TREMOR
-    ifeq ($(TARGET_ARCH_ABI),armeabi)
-	LOCAL_CFLAGS += -D_ARM_ASSEM_
-    endif
-    ifeq ($(TARGET_ARCH_ABI),armeabi-v7a)
-	LOCAL_CFLAGS += -D_ARM_ASSEM_
-    endif
+  #   ifeq ($(TARGET_ARCH_ABI),armeabi)
+	# LOCAL_CFLAGS += -D_ARM_ASSEM_
+  #   endif
+  #   ifeq ($(TARGET_ARCH_ABI),armeabi-v7a)
+	# LOCAL_CFLAGS += -D_ARM_ASSEM_
+  #   endif
     LOCAL_SRC_FILES += \
         $(VORBIS_LIBRARY_PATH)/mdct.c \
         $(VORBIS_LIBRARY_PATH)/block.c \
diff -r d9e357c04dfa external/smpeg2-2.0.0/video/proto.h
--- a/external/smpeg2-2.0.0/video/proto.h	Sun Mar 23 09:51:24 2014 -0700
+++ b/external/smpeg2-2.0.0/video/proto.h	Tue Apr 01 18:45:55 2014 +0200
@@ -200,6 +200,7 @@
    double *floatClockTime));
 
 
+#ifndef __ANDROID__
 #ifndef NOCONTROLS
 /* ctrlbar.c */
 double StopWatch P((int action ));
@@ -211,5 +212,6 @@
 void ControlBar P((VidStream **vid_stream, XInfo *xinfo, int numMovies ));
 void ControlLoop P((VidStream **theStream, XInfo *xinfo, int numStreams ));
 #endif /* !NOCONTROLS */
+#endif /* !__ANDROID__ */
 
 #undef P
diff -r d9e357c04dfa external/smpeg2-2.0.0/video/util.cpp
--- a/external/smpeg2-2.0.0/video/util.cpp	Sun Mar 23 09:51:24 2014 -0700
+++ b/external/smpeg2-2.0.0/video/util.cpp	Tue Apr 01 18:45:55 2014 +0200
@@ -53,9 +53,11 @@
 #include "video.h"
 #include "proto.h"
 #include "util.h"
+#ifndef __ANDROID__
 #ifndef NOCONTROLS
 #include "ctrlbar.h"
 #endif
+#endif
 
 /*
    Changes to make the code reentrant:
